proto/storage/resource_collection.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+
+option go_package = "storage";
+option java_package = "io.stackrox.proto.storage";
+
+import "google/protobuf/timestamp.proto";
+import "gogoproto/gogo.proto";
+import "storage/user.proto";
+import "storage/policy.proto";
+
+package storage;
+
+message ResourceCollection {
+  message EmbeddedResourceCollection {
+    // this is searchable to force a separate table
+    string id = 1 [(gogoproto.moretags) = 'search:"Embedded Collection ID",hidden'];
+  }
+
+  string id                                                = 1 [(gogoproto.moretags) = 'sql:"pk"']; // primary key
+  string name                                              = 2 [(gogoproto.moretags) = 'search:"Collection Name"'];
```

Comment:
Mark `name` as `unique`. We must enforce Collections to have unique names otherwise the workflow would become utterly confusing.